dev/bots/test.sh

```diff
@@ -1,20 +1,27 @@
 #!/bin/bash
-set -ex
 
 export PATH="$PWD/bin:$PWD/bin/cache/dart-sdk/bin:$PATH"
 
-die() {
+trap detect_error_on_exit EXIT HUP INT QUIT TERM
+
+detect_error_on_exit() {
+    exit_code=$?
     set +x
-    echo "Error: script exited early due to error"
-    exit 1
+    if [[ $exit_code -ne 0 ]]; then
+        echo "--------------------------------------------"
+        echo "Error: script exited early due to error ($exit_code)"
+        echo "--------------------------------------------"
```

Comment:
Rather than hyphens, consider using "━", the U+2501 BOX DRAWINGS HEAVY HORIZONTAL character. Maybe combined with 😞 U+1F61E DISAPPOINTED FACE. See https://github.com/flutter/flutter/blob/master/packages/flutter_test/lib/src/widget_tester.dart#L92 or https://github.com/flutter/flutter/blob/master/packages/flutter_tools/lib/src/usage.dart#L103 for inspiration.